File: GymApp/composeApp/src/commonMain/kotlin/com/gym/gymapp/ui/screens/MainScreen.kt
package com.gym.gymapp.ui.screens

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.runtime.*
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import cafe.adriel.voyager.core.screen.Screen
import cafe.adriel.voyager.navigator.LocalNavigator
import cafe.adriel.voyager.navigator.currentOrThrow
import com.gym.gymapp.ui.components.GymBottomBar
import com.gym.gymapp.ui.components.MainTab
import com.gym.gymapp.ui.navigation.Screens
import com.gym.gymapp.ui.screens.attendance.AttendanceScreen
import com.gym.gymapp.ui.screens.dashboard.DashboardScreen
import com.gym.gymapp.ui.screens.members.MemberListScreen
import com.gym.gymapp.ui.screens.profile.ProfileScreen
import com.gym.gymapp.ui.screens.payments.PaymentsScreen
import com.gym.gymapp.ui.screens.plans.PlansScreen
import com.gym.gymapp.ui.screens.announcements.BroadcastScreen


import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Add
import androidx.compose.material3.FloatingActionButton
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import com.gym.gymapp.ui.theme.Black

import com.gym.gymapp.ui.theme.Black
import com.gym.gymapp.AppBackHandler
import com.gym.gymapp.ui.components.NotificationManager
import com.gym.gymapp.ui.components.AppNotificationType
import kotlinx.coroutines.delay

import androidx.compose.runtime.saveable.rememberSaveable
import androidx.compose.ui.Alignment
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp

class MainScreen : Screen {
    @Composable
    override fun Content() {
        var selectedTab by rememberSaveable { mutableStateOf(MainTab.HOME) }
        val navigator = LocalNavigator.currentOrThrow
        val scope = rememberCoroutineScope()
        var lastBackPressTime by remember { mutableStateOf(0L) }

        AppBackHandler(enabled = true) {
            if (navigator.canPop) {
                navigator.pop()
            } else if (selectedTab != MainTab.HOME) {
                selectedTab = MainTab.HOME
            } else {
                val currentTime = com.gym.gymapp.getPlatform().currentTimeMillis()
                if (currentTime - lastBackPressTime < 2000) {
                    com.gym.gymapp.getPlatform().exitApp()
                } else {
                    lastBackPressTime = currentTime
                    NotificationManager.showNotification("Press again to exit", AppNotificationType.INFO)
                }
            }
        }

        Scaffold(
            bottomBar = {
                GymBottomBar(
                    selectedTab = selectedTab,
                    onTabSelected = { selectedTab = it }
                )
            },
            floatingActionButton = {
                if (selectedTab == MainTab.MEMBERS || selectedTab == MainTab.PLANS) {
                    FloatingActionButton(
                        onClick = { 
                            if (selectedTab == MainTab.MEMBERS) navigator.push(Screens.AddMember())
                            else navigator.push(Screens.AddPlan())
                        },
                        containerColor = MaterialTheme.colorScheme.primary,
                        contentColor = MaterialTheme.colorScheme.onPrimary,
                        shape = CircleShape
                    ) {
                        Icon(Icons.Default.Add, contentDescription = "Add Item")
                    }
                }
            },

            containerColor = MaterialTheme.colorScheme.background // Restore theme support
        ) { padding ->
            Box(modifier = Modifier.padding(padding)) {
                when (selectedTab) {
                    MainTab.HOME -> DashboardScreen(
                        onAddMember = { navigator.push(Screens.AddMember()) },
                        onViewMembers = { selectedTab = MainTab.MEMBERS },
                        onViewAttendance = { selectedTab = MainTab.PAYMENTS }, // Linked to Payments for now
                        onViewProfile = { selectedTab = MainTab.PROFILE },
                        onMemberClick = { id -> navigator.push(Screens.MemberDetails(id)) }
                    )
                    MainTab.MEMBERS -> MemberListScreen(
                        onBack = { selectedTab = MainTab.HOME },
                        onMemberClick = { id -> navigator.push(Screens.MemberDetails(id)) },
                        onAddMember = { navigator.push(Screens.AddMember()) }
                    )
                    MainTab.PAYMENTS -> {
                        PaymentsScreen()
                    }
                    MainTab.PLANS -> {
                        PlansScreen(
                            onAddPlan = { navigator.push(Screens.AddPlan()) },
                            onEditPlan = { navigator.push(Screens.AddPlan()) }
                        )
                    }


                    MainTab.PROFILE -> ProfileScreen(
                        onBack = { selectedTab = MainTab.HOME },
                        onLogout = { 
                            navigator.replaceAll(Screens.Login())
                        }
                    )
                }
            }
        }
    }
}


@Composable
fun PlaceholderScreen(title: String, description: String) {
    Column(
        modifier = Modifier.fillMaxSize().padding(24.dp),
        verticalArrangement = Arrangement.Center,
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Text(
            text = title, 
            style = MaterialTheme.typography.headlineMedium, 
            fontWeight = FontWeight.Bold,
            color = MaterialTheme.colorScheme.onBackground
        )
        Spacer(Modifier.height(8.dp))
        Text(
            text = description, 
            style = MaterialTheme.typography.bodyMedium,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
            textAlign = androidx.compose.ui.text.style.TextAlign.Center
        )
    }
}

File: gym_flutter_app/lib/ui/screens/main_screen.dart
import 'package:flutter/material.dart';
import '../widgets/gym_bottom_bar.dart';
import 'dashboard_screen.dart';
import 'member_list_screen.dart';
import 'profile_screen.dart';

class MainScreen extends StatefulWidget {
  const MainScreen({super.key});

  @override
  State<MainScreen> createState() => _MainScreenState();
}

class _MainScreenState extends State<MainScreen> {
  MainTab _selectedTab = MainTab.home;

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);

    return Scaffold(
      backgroundColor: theme.colorScheme.surface,
      body: _buildBody(),
      bottomNavigationBar: GymBottomBar(
        selectedTab: _selectedTab,
        onTabSelected: (tab) => setState(() => _selectedTab = tab),
      ),
      floatingActionButton: (_selectedTab == MainTab.members || _selectedTab == MainTab.plans)
          ? FloatingActionButton(
              onPressed: () {
                // Handle Add Member or Plan
              },
              backgroundColor: theme.colorScheme.primary,
              foregroundColor: Colors.black,
              shape: const CircleBorder(),
              child: const Icon(Icons.add),
            )
          : null,
    );
  }

  Widget _buildBody() {
    switch (_selectedTab) {
      case MainTab.home:
        return const DashboardScreen();
      case MainTab.members:
        return const MemberListScreen();
      case MainTab.payments:
        return const _PlaceholderScreen(title: "Payments", description: "Payment history and management");
      case MainTab.plans:
        return const _PlaceholderScreen(title: "Membership Plans", description: "Manage your gym membership plans");
      case MainTab.profile:
        return const ProfileScreen();
    }
  }
}

class _PlaceholderScreen extends StatelessWidget {
  final String title;
  final String description;

  const _PlaceholderScreen({required this.title, required this.description});

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    return Center(
      child: Padding(
        padding: const EdgeInsets.all(24.0),
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            Text(
              title,
              style: theme.textTheme.headlineMedium?.copyWith(fontWeight: FontWeight.bold, color: theme.colorScheme.onSurface),
            ),
            const SizedBox(height: 8),
            Text(
              description,
              style: theme.textTheme.bodyMedium?.copyWith(color: theme.colorScheme.onSurface.withValues(alpha: 0.6)),
              textAlign: TextAlign.center,
            ),
          ],
        ),
      ),
    );
  }
}
